src/App.tsx

```diff
@@ -17,18 +56,36 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onChange={setSelectValue}
+                onSearch={setQuery}
+                selectValue={value}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {visibleTodos[0] === defaultTodo
+                ? <Loader />
+                : (
+                  <TodoList
+                    todos={searchedTodos}
+                    onClick={handleClick}
+                    selectedTodoId={selectedTodo.id}
+                  />
+                )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {isModalVisible && (
```

Comment:
you don't need `isModalVisible` - you can check `selectedTodo` (if default value === null)